main.c

```diff
@@ -431,6 +432,34 @@ uint32_t parse_color(const char *color) {
 	return res;
 }
 
+static void print_formatted_result(const struct slurp_box result, const char *format) {
+	for (size_t i = 0; format[i] != 0; i++) {
+		char c = format[i];
+		if (c == '%') {
+			char next = format[i + 1];
+			
+			i++; // Skip the next character (x, y, w or h)
+			switch (next) {
+			case 'x':
+				printf("%u", result.x);
+				continue;
```

Comment:
`continue` is a little bit weird here. Can we use `break` instead?

Reply:
Ah, I see. `continue` won't affect the `for` loop, since it's wrapped in a `switch`. Probably need to use a `default` case.

Reply:
We're continuing to keep the `i--` from happening and the character from being printed. I could break and put `printf("%c", c)` in an else-block, remove 'i--` and put `i++` in the case blocks instead of before the switch, but that's just more lines and repetition, isn't it?

Reply:
Eh, I see that `continue` does affect the loop instead of the `switch`. It's pretty confusing, but I guess it makes sense.

I'm fine with this approach.